File: Data Structures & Algorithms/depth-of-binary-tree/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    int maxH(TreeNode* node){
        if (node == NULL){ return 0;}

        int lh = maxH(node->left);
        int rh = maxH(node->right);

        if (lh>rh) {return lh+1;}
        else{ return rh+1;}
    }

    int maxDepth(TreeNode* root) {
        return maxH(root);
    }
};

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if root == None:
            return []
        q = [root]
        level = 0
        ans = []

        while q:
            for i in range(len(q)):
                node = q.pop(0)

                if node.left != None:
                    q.append(node.left)

                if node.right != None:
                    q.append(node.right)
            ans.append(node.val)
        return ans